### PR TITLE
User pets endpoint

### LINKED ISSUE #16: GET Users pets endpoint
API/V1/users/:id/pets
returns users pet objects or empty hash

### PR DESCRIPTION
**Pull request by jakejakearell**
# Modifies
- `app/models/users`: Add a method for getting user pets
- `spec/models/user_spec`: Add a test for 
- `config/routes`: Add route for endpoint

# Adds
- `app/controllers/api/v1/user_pets_controller`: Serialize user pet data 
- `spec/requests/api/v1/user_pets_request`: tests the data structure of JSON from user_pets endpoint, edge case testing.

<details>
<summary>Issues</summary>
  closes #16 
<br>

</details>

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -5,6 +5,10 @@
     namespace :v1 do
       resources :users, only: [:create, :index]
       resources :pets, only: :create
+
+      resources :users, only: [:index] do
+        resources :pets, controller: :user_pets, only: :index
+      end
```
Comment (arikalea) [inline config/routes.rb:11]: We may need to DRY this up later on

File: spec/requests/api/v1/user_pets_request_spec.rb [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+      expect(response.status).to eq(404)
+      expect(user_pets).to be_a(Hash)
+      expect(user_pets.count).to eq(2)
+      expect(user_pets).to have_key(:error)
+      expect(user_pets[:error]).to eq("No such user")
+      expect(user_pets).to have_key(:status)
+      expect(user_pets[:status]).to eq(404)
+    end
+  end
+end
```
Comment (arikalea) [inline spec/requests/api/v1/user_pets_request_spec.rb:88]: This looks great!

**Review by arikalea — APPROVED** [review]
